### ISSUE TITLE
AirWatch Detects Jailbreak

### ISSUE THREAD

**Issue by IsaiahJTurner** [n00neimp0rtant/xCon-Issues#328]
To be clear, this is NOT an "I have iOS 7 and it doesn't work!" issue. I have tested it with the latest version of the app on 3 iOS 6 devices to confirm. It has not worked for a while but before now, there was not an iOS 7 jailbreak so the community has slowed down.

Through my Burp Suite adventure I discovered a few things. I grabbed each of the IPA files for all previous versions and started working backwards to see when things went wrong hoping that there were unpatched things before to give me clues. I discovered that prior to version (and this may be slightly off as I have not checked recently) 4.1 they sent a JSON request unecrypted that included an "isCompromised" key and a value of either 0 or 1. This can no longer be overwritten because I think that this data is now encrypted. Not quite sure what encryption, I tired Base64 and got a small amount of readable data (the line reads "    0   1I0E0-0(1&0$UAirWatch Device Services Root0
" making me believe that it uses NSData but that did not seem to work either when I built a little app in Obj-C. Granted, I really don't know much about NSData so if you want to take a look at this whole string I can send it to you (just don't want to put it online). I do however know that the data is still in the same format because the app is nice enough to provide a detailed log.

I tried doing some stuff in Plex, some of that was needed to bypass the SSL validation until I discovered SSL Kill Switch (A tweak on GitHub) that made life allot easier.
This is the beacon that contains the "isCompromised" status.
`POST /deviceservices/SecureChannel.aws HTTP/1.1
Host: ds###.awmdm.com
Proxy-Connection: keep-alive
Accept-Encoding: gzip, deflate
Content-Type: application/json
Accept-Language: en-us
Accept: _/_
Content-Length: 3679
Connection: keep-alive
x-aw-sdk-version: 3.3
User-Agent: AirWatch/4.8.888 CFNetwork/672.0.8 Darwin/14.0.0

<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
    <key>beacon</key>
    <string>Base64/NSData/Some other unreadable data</string>
    <key>bundleId</key>
    <string>com.air-watch.agent</string>
    <key>deviceType</key>
    <string>2</string>
    <key>uid</key>
    <string>Some string of numbers and letters that I assume is entirely unafiliated with IOAESEncrytion</string>
</dict>
</plist>`

I cannot seem to find out how the detection works in anything I try.
Sorry if anything I say doesn't make sense. I might not have the terminology on some things rite. I might have gotten some things entirely wrong so I would love your feedback. Also sorry for making this more of a reverse engineering a specific app post verse what xCon usually does which is patch the source and not the specific apps. I just could not seem to find anyone who had a clue what this stuff meant and the few people I found couldn't help because they were busy.

**Comment by jws43** [comment]
I have Airwatch v4.9.3 and being detected but still allowing access. Going to find out when IT guy is back next week if going to be allowed to keep my jailbreak but would be ideal if xCon could bypass detection entirely.

**Comment by nullogy** [comment]
Does anyone know if this method is still working? I tried to reach out to Isiah with no luck.

**Comment by colereynolds** [comment]
Yes. I'm doing it now on iOS 9. Restore your iPhone to new, enroll in AirWatch, delete the AirWatch app, then jailbreak.

**Comment by nullogy** [comment]
If I restore right now, it'll upgrade the phone to 9.1 and then I can't JB. My current status is iOS 9.0.1, jailbroken with AirWatch installed. What do you think my options could be?

Thanks!

**Comment by leelouch** [comment]
too late ! u need to wait for the next jailbreak windows ! 
But I confirm it still working with 9.0.1

**Comment by nullogy** [comment]
I have no other option in the mean time, correct?

**Comment by leelouch** [comment]
no option ! but ... 
I may have an option for u , but it is risky.
Saurik released a tool called cydia impactor that remove jailbreak without restoring your iphone! I do not know if it is released for 9.0x . you may need to wait a little bit.
So the idea is to remove the jailbreak . then enroll your device and than jailbreak :) 
it may work . but if for a reason the re jailbreak fail , u will need to restore an update to latest version . 
good luck

**Comment by colereynolds** [comment]
Could you not just remove the AirWatch app and be good to go? Or has it not installed your profiles yet?

**Comment by nullogy** [comment]
@colereynolds, I have the AirWatch app installed with the profiles. After my device was jailbroken. I think my only option right now would be to see if the method that @leelouch mentioned works.

**Comment by colereynolds** [comment]
I guess I'm confused on what your issue is... If you uninstall the AirWatch app, it won't report that your device is compromised anymore. The only reason you need to use Cydia Impactor is if it wouldn't install the profiles or if the jailbreak detection is preventing you from doing something..

**Comment by nullogy** [comment]
@colereynolds, can you message via twitter if you have one? I can be reached at @TheMBAer.

**Comment by colereynolds** [comment]
Just followed you. Follow me back, and I'll DM you tomorrow.
